gitCoreApi/src/main/java/com/virtuslab/gitcore/api/IGitCoreRepository.java

```diff
@@ -10,11 +10,15 @@
 
   IGitCoreLocalBranch getLocalBranch(String branchName) throws GitCoreException;
 
-  IGitCoreRemoteBranch getRemoteBranch(String branchName) throws GitCoreException;
+  Option<IGitCoreRemoteBranch> getRemoteBranch(String branchName, String remoteName) throws GitCoreException;
```

Comment:
It looks it's unused outside of `GitCoreRepository` and TBH looks like sth that shouldn't be exposed in the interface

Reply:
> looks like sth that shouldn't be exposed in the interface

Why?

Reply:
Ok, actually pls ignore this point; you're right, we'll need to be able to seek remote branches by name... given that most people don't to `git push -u`, tracking info notoriously tends to remain unset. We'll need to guess the remote name just basing on branch name and not on git's tracking info... actually, creating an issue for that

Reply:
#267